cmd/generic-handlers.go

```diff
@@ -469,3 +474,68 @@ func (h pathValidityHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 	h.handler.ServeHTTP(w, r)
 }
+
+// logginHandler logs all HTTP requests and responses
+// and write the log to a io.Writer object.
+type loggingHandler struct {
+	writer     io.Writer
+	handler    http.Handler
+	timeFormat string
+}
+
+func (h loggingHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Time that will be printed in log
+	timeFormat := "2006-01-02 15:04:05 -0700"
+	// Log HTTP request
+	requestDump, err := httputil.DumpRequest(r, false)
+	if err == nil {
+		reqTime := time.Now().Format(timeFormat) + " >> "
+		h.writer.Write([]byte(reqTime))
+		h.writer.Write(requestDump)
+		h.writer.Write([]byte("\n"))
+	} else {
+		errorIf(err, "Cannot dump HTTP request")
+	}
+
+	// Setup a http recorder to save HTTP response
+	rec := httptest.NewRecorder()
+	h.handler.ServeHTTP(rec, r)
+
+	// Log HTTP Response
+	respDump, err := httputil.DumpResponse(rec.Result(), false)
+	if err == nil {
+		respTime := time.Now().Format(timeFormat) + " << "
```

Comment:
When parallel request/response are logged, the relation between one request and its response it not properly related here.

You would need to create a request ID kind of relation (I am OK this as `time.Now().Format(timeFormat)`) when dumping response.